[Bugfix] Fix weight_loader v1 block scale

kyuyeunk opened the pull request:
## Purpose

Fix a bug in linear layer weight_loader v1 where `shard_size` is incorrectly calculated if parameter the is `BlockQuantScaleParameter`

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

yaochengji reviewed (commented):
Thanks for the contribution! Is there a way to add a unit test for this fix?

kyuyeunk committed:
[Bugfix] Fix weight_loader v1 block scale

Signed-off-by: Kyuyeun Kim <kyuyeunk@google.com>

kyuyeunk commented:
/gemini review

@codex review

kyuyeunk commented:
> Thanks for the contribution! Is there a way to add a unit test for this fix?

Hi @yaochengji, I have addressed all the comments. w.r.t unit tests, I've tried to follow a similar approach as other helper functions in the file but they do not exist either: https://github.com/vllm-project/vllm/blob/99dcf5dcc5946168074cac632b9f5bb89a92775e/vllm/model_executor/layers/linear.py#L67-L95

Do you have any recommendations on how to approach this? I think more fundamental issue is that there doesn't seem to be a comprehensive unit test for `weight_loader` and all the tests are done with `weight_loader_v2`. And tpu-inference code path utilizes `weight_loader`.

yaochengji reviewed (approved):
LGTM, thanks for the fix!
